lib/bundler/cli/doctor.rb

```diff
@@ -90,5 +92,37 @@ def run
         Bundler.ui.info "No issues found with the installed bundle"
       end
     end
+
+  private
+
+    def check_home_permissions
+      check_for_files_not_owned_by_current_user_but_still_rw
+      check_for_files_not_readable_or_writable
+    end
+
+    def check_for_files_not_owned_by_current_user_but_still_rw
+      return unless any_files_not_owned_by_current_user_but_still_rw?
+      Bundler.ui.warn "Files exist in Bundler home that are owned by another " \
+        "user, but are stil readable/writable"
+    end
+
+    def check_for_files_not_readable_or_writable
+      return unless any_files_not_readable_or_writable?
+      raise ProductionError, "Files exist in Bundler home that are not " \
```

Comment:
I'm chose to raise```ProductionError``` because I saw it being raised in the ```run``` method; if I should raise a different error please let me know.

Reply:
I don't agree with using exceptions as part of regular program behavior flow. I suggest printing a message to the user using `Bundler.ui.info` instead.